pkg/observ/observ.go

```diff
@@ -40,16 +87,37 @@ func RegisterTraceExporter(URL, service, ENV string) (*(jaeger.Exporter), error)
 	})
 
 	if err != nil {
-		return nil, errors.E(op, err)
+		return errors.E(op, err)
 	}
 
 	// And now finally register it as a Trace Exporter
-	trace.RegisterExporter(je)
+	trace.RegisterExporter(j.exporter)
```

Comment:
let's move this duplication to a private function multiple implementations can use: 

```golang
func registerExporter(ex trace.Exporter) {
  trace.RegisterExporter(ex)
  if ENV == "development" {...}
}
```

Reply:
Hey, how do you reckon I solve this? I was thinking of creating a new interface method to register the tracer.

Reply:
@bhaveshpraveen just like the function I have above. The `trace.Exporter` argument will accept both datadog and other exporters. 

Now that I took a second look, you actually dont need the Exporter interface in the observ package. The `RetrieveExporter` can just be `RegisterExporter` and it does all the registration for you, and just returns a `func()` that handles the "flushing" or "stopping"

Reply:
@marwan-at-work : I think I understand what you're saying
So, the code would look something like this?
```
function_to_flush, err := RegisterExporter(...)
function_to_flush()
```